File: lc_py/lcPy/array/minCosts3502.py
class MinCosts:
    def minCosts(self, cost: List[int]) -> List[int]:
        for i in range(1, len(cost)):
            cost[i] = min(cost[i], cost[i - 1])
        return cost



if __name__ == "__main__":

    sol = MinCosts()
    cost = [5,3,1,2,4]
    print(sol.minCosts(cost))

File: lc_go/lcGo/array/min Costs 3502.go
package main

import "fmt"

func minCosts(cost []int) []int {
    for i:= 1;i < len(cost);i++ {
        cost[i] = min(cost[i - 1],cost[i])
    }
    return cost
}
func main(){
	cost := []int{5,3,1,2,4}
	res := minCosts(cost)
	fmt.Println(res)
}